tensorflow/core/kernels/mkl_conv_ops.cc

```diff
@@ -1590,19 +1588,17 @@ class MklQuantizedConv2DOp
       // 1. Bias is not const;
       // 2. Bias is const, but bias cache is empty (first iteration).
 
-      // TODO(intel-tf): Re-enable bias caching. Currently, the graph obtained
-      // after quantize_graph.py does not run with correct accuracy with this
-      // feature enabled.
-      is_bias_const_ = false;
-      if (!is_bias_const_ || IsBiasCacheEmpty(context)) {
-        size_t depth = min_filter_vector.NumElements();
-        std::vector<float> scales(depth);
-        for (size_t i = 0; i < depth; ++i) {
-          scales[i] =
-              int_const_scale_limit /
-              (std::max(std::abs(max_input), std::abs(min_input)) *
-               std::max(std::abs(max_filter[i]), std::abs(min_filter[i])));
-        }
+      size_t depth = min_filter_vector.NumElements();
+      std::vector<float> scales(depth);
+      for (size_t i = 0; i < depth; ++i) {
+        scales[i] =
+            int_const_scale_limit /
+            (std::max(std::abs(max_input), std::abs(min_input)) *
+             std::max(std::abs(max_filter[i]), std::abs(min_filter[i])));
+      }
+      if (!is_bias_const_ || IsBiasCacheEmpty(context) ||
```

Comment:
To make the code easier to understand, could you please restructure it this way?

- Check if bias is enabled at the beginning of the function (`GetBiasHandle`) and return early if so. 
```c++
if (!bias_enabled) return nullptr;
```
- Take the rest of the code out of the current `if (bias_enabled)`.
- Return the `GetCachedBias` case first, checking if scales are valid on the go (not storing in vector). Something like this:
```c++
// If scales_ exists, checks if the current scales are the same.
bool scales_are_valid = (depth == scales_.size());
const float intermediate_scale_limit = 
    int_const_scale_limit / (std::max(std::abs(max_input), std::abs(min_input));
for (size_t i = 0; scales_are_valid && i < depth; ++i) {
  float multiplier = std::max(std::abs(max_filter[i]), std::abs(min_filter[i]));
  if (std::abs(intermediate_scale_limit - scales_[i] * multiplier) > some_epsilon) {
    scales_are_valid = false;
  }
}
// The bias has been cached and is safe to use.
if (is_bias_const_ && !IsBiasCacheEmpty(context) && scales_are_valid) {
  return GetCachedBias(context);
}
// The rest.
```

Reply:
@penpornk As I did not know the scales are valid until I make sure all the scales are some to the original one. So I just can stop to compare the two float if the scales already invalid. I also need to compute all the scales and set to the new one.
Do you have any comments here, thanks.